src/drt/src/dr/FlexGridGraph.h

```diff
@@ -751,37 +751,46 @@ class FlexGridGraph
     if (x2 < x1 || y2 < y1) {
       return;
     }
-    if (getZDir(z) == Horizontal) {
-      for (int i = y1; i <= y2; i++) {
-        auto idx1 = getIdx(x1, i, z);
-        auto idx2 = getIdx(x2, i, z);
-        std::fill(guides_.begin() + idx1, guides_.begin() + idx2 + 1, 1);
-      }
-    } else {  // == Vertical
-      for (int i = x1; i <= x2; i++) {
-        auto idx1 = getIdx(i, y1, z);
-        auto idx2 = getIdx(i, y2, z);
-        std::fill(guides_.begin() + idx1, guides_.begin() + idx2 + 1, 1);
-      }
+    switch(getZDir(z)) {
+      case dbTechLayerDir::HORIZONTAL:
+        for (int i = y1; i <= y2; i++) {
+          auto idx1 = getIdx(x1, i, z);
+          auto idx2 = getIdx(x2, i, z);
+          std::fill(guides_.begin() + idx1, guides_.begin() + idx2 + 1, 1);
+        }
+        break;
+      case dbTechLayerDir::VERTICAL:
+        for (int i = x1; i <= x2; i++) {
+          auto idx1 = getIdx(i, y1, z);
+          auto idx2 = getIdx(i, y2, z);
+          std::fill(guides_.begin() + idx1, guides_.begin() + idx2 + 1, 1);
+        }
+        break;
+      default: {}
```

Comment:
case dbTechLayerDir::NONE

is better in case a new value is ever added.

Reply:
@maliberty what about this? It's ultimately similar handling to before and doesn't leave uncovered cases.
```
    switch(getZDir(z)) {
      case dbTechLayerDir::HORIZONTAL:
        //....
        break;
      default:
        // TODO replace with logger
        cout << "WARNING: layer " << z << "Has unsupported TechLayerDir "
             << getZDir(z).getString() << ". Assuming VERTICAL."
      case dbTechLayerDir::VERTICAL:
        //...
        break;
```

Reply:
IMO this whole thing should get moved to the .cpp later.

Reply:
I saying that all cases should be explicitly listed rather than using default.

Reply:
I can do that, My thinking was to have a runtime warning rather than a compiler warning for uncovered case. It should be noted that the old code simply assumed everything not horizontal was vertical.